datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -233,42 +233,60 @@ impl LogicalPlan {
     /// logical plan node. This does not include expressions in any
     /// children
     pub fn expressions(self: &LogicalPlan) -> Vec<Expr> {
+        let mut exprs = vec![];
+        self.observe_expressions(|e| exprs.push(e.clone()));
```

Comment:
The basic idea is to refactor the current code to clone all expressions to just call a function